tools/kura_inspector/src/main.rs

```diff
@@ -67,11 +67,33 @@ fn main() {
     clippy::expect_fun_call
 )]
 fn print_blockchain(block_store_path: &Path, from_height: u64, block_count: u64) {
+    let mut _internal_path_buf_for_modified_block_store_path;
+
+    let block_store_path = match block_store_path.file_name() {
+        Some(os_str_file_name) => {
+            let file_name_str = os_str_file_name.to_str().unwrap_or("");
+            if file_name_str == "blocks.data" || file_name_str == "blocks.index" {
+                _internal_path_buf_for_modified_block_store_path = block_store_path.to_path_buf();
+                _internal_path_buf_for_modified_block_store_path.pop();
+                &_internal_path_buf_for_modified_block_store_path
+            } else {
+                block_store_path
+            }
+        }
+        None => block_store_path,
+    };
+
     let block_store = StdFileBlockStore::new(block_store_path);
```

Comment:
I think this peace of code is a good place for `Cow`, we ether return reference as is or modify.

```suggestion
    let mut block_store_path: std::borrow::Cow<'_, Path> = block_store_path.into();

    if let Some(os_str_file_name) = block_store_path.file_name() {
        let file_name_str = os_str_file_name.to_str().unwrap_or("");
        if file_name_str == "blocks.data" || file_name_str == "blocks.index" {
            block_store_path.to_mut().pop();
        }
    }

    let block_store = StdFileBlockStore::new(&block_store_path);
```